engine/resource/src/resource.cpp

```diff
@@ -1081,17 +1085,28 @@ static Result DoGet(HFactory factory, const char* name, void** resource)
             return RESULT_UNKNOWN_RESOURCE_TYPE;
         }
 
-        void *buffer;
-        uint32_t file_size;
-        Result result = LoadResource(factory, canonical_path, name, &buffer, &file_size);
+        void* message = 0;
+        void* buffer = 0;
+        uint32_t buffer_size = 0;
+
+        GetResourceDataCallbackParams get_resource_data_params = {};
+        get_resource_data_params.m_Factory                     = factory;
+        get_resource_data_params.m_CanonicalPath               = canonical_path;
+        get_resource_data_params.m_Name                        = name;
+        get_resource_data_params.m_Message                     = &message;
+        get_resource_data_params.m_Buffer                      = &buffer;
+        get_resource_data_params.m_BufferSize                  = &buffer_size;
+        get_resource_data_params.m_UserData                    = userdata;
+
+        Result result = get_resource_data_cb(&get_resource_data_params);
         if (result != RESULT_OK) {
             if (result == RESULT_RESOURCE_NOT_FOUND) {
                 dmLogWarning("Resource not found: %s", name);
             }
             return result;
         }
 
-        assert(buffer == factory->m_Buffer.Begin());
+        assert(buffer || message);
```

Comment:
Please make it two separate asserts.
If it fails, we won't know which one was wrong.